src/libostree/ostree-fetcher-curl.c

```diff
@@ -330,7 +331,14 @@ check_multi_info (OstreeFetcher *fetcher)
             }
           else
             {
-              g_task_return_new_error (task, G_IO_ERROR, G_IO_ERROR_FAILED,
+              /* When it is not a file and the OSTREE_CURL_IO_RETRY
+               * environment variable is set, we want to retry the request.
+               * We accomplish that by using G_IO_ERROR_TIMED_OUT.
+               */
+              should_retry = g_getenv ("OSTREE_CURL_IO_RETRY") != NULL;
```

Comment:
General style in ostree is to use declare-and-initialize where possible, so this can be `gboolean should_retry = ...` instead of declaring above.